llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -612,6 +612,48 @@ void getAddressSanitizerParams(const Triple &TargetTriple, int LongSize,
   *OrShadowOffset = Mapping.OrShadowOffset;
 }
 
+void removeASanIncompatibleFnAttributes(Function &F, bool ReadsArgMem) {
+  // Remove memory attributes that are invalid with ASan and HWSan.
+  // ASan checks read from shadow, which invalidates memory(argmem: *)
+  // Short granule checks on function arguments read from the argument memory
+  // (last byte of the granule), which invalidates writeonly.
```

Comment:
This needs reframing in terms of ReadsArgMem.

Reply:
Better described the parameter in the common header and explained the implementation detail in `HWAddressSanitizer.cpp` because its specific to that sanitizer.